### PR TITLE
Wip 0.0.8

### PR DESCRIPTION
**Pull request by patricksimonian**
- improved docs
- added `.isA` static method to TypeCheck
- added units tests for TypeCheck
- builds are unminified to allow for better debugging
- add helper method to ImplicitAuthManager to return a bearer header
- add kc_idp_hint flag in option
- update documentation

### CODE REVIEW

File: __tests__/TypeCheck.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    expect(TypeCheck.isString(badValue)).toBe(false);
+  });
+
+  it('checks for Boolean', () => {
+    const value = true;
+    const badValue = 9;
+    expect(TypeCheck.isBoolean(value)).toBe(true);
+    expect(TypeCheck.isBoolean(badValue)).toBe(false);
+  });
+  test('isA throws if object contructor is invalid', () => {
```
Comment (jleach) [inline __tests__/TypeCheck.test.js]: Did you mean `it` rather than `test`?

**Comment by jleach** [comment]
I think that, in general, starting numbering with `0.1.0` is good practices. Usually the last digit is meant for minor releases. Right now, every change you make looks like a bug fix or tweak. By using 0.1.0 you can do bit release and small.

https://semver.org

**Comment by patricksimonian** [comment]
Great point, i'll make those changes including the versioning change and stick with semver. I'll look to get that `kc_idp_hint` flag in the code for this release as well. Thanks for reviewing

**Comment by patricksimonian** [comment]
after doing some research there is nothing saying the the key cloak docs that kc_idp_hint will work with implicit authorization. It only mentions authorization. I'm going to test that out before i make any changes. The one concern i have is that this flag is strictly tied to key cloak, it begs for the implicit auth manager to take in a plugin that has configurations for how urls are built so that it's not strictly tied to being an auth manager of key cloak apps.. something to think about

**Pull request merged**